### PR TITLE
Added test cases with validate authority false

### PR DESCRIPTION
**Pull request by kreedula**

### CODE REVIEW

**Comment by kreedula** [comment]
@shoatman These tests doesn't reproduce the null pointer issue we discussed about. Had an offline discussion with Brian and he mentioned the issue is not reproducible. Let me know what you think

File: automation/src/test/java/com/microsoft/identity/common/test/automation/AcquireTokenAuthorityTest.java [hunk 1]
```diff
@@ -121,12 +125,37 @@ private static User getUser(TestConfigurationQuery query) {
...
+
+    // invalid authority test, no validation but should exit gracefully
+    @Test
+    public void should_be_able_to_exit_silently_on_validate_authority_false_for_invalid_authority(){
+        james.attemptsTo(
+                clearCache,
+                clickDone,
+                acquireToken
+                        .validateAuthority(false)
+                        .withAuthority("https://www.cnn.com/world/"));
```
Comment (iambmelt) [inline automation/src/test/java/com/microsoft/identity/common/test/automation/AcquireTokenAuthorityTest.java]: We should change this value to use `contoso` or another Microsoft-owned domain value

**Comment by iambmelt** [comment]
The issue discovered in testing was that [`AzureActiveDirectory#sAadClouds` ](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectory.java#L50)was empty when [`false == validateAuthority`](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/dev/adal/src/main/java/com/microsoft/aad/adal/AuthenticationContext.java#L100).

For either a real or mock authority, can this test be expanded to validate that [`AzureActiveDirectory#hasCloudHost(URL)`](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectory.java#L57-L59) contains the provided authority as an [`AzureActiveDirectoryCloud`](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryCloud.java) instance and that [`false == isValidated()`](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/blob/dev/common/src/main/java/com/microsoft/identity/common/internal/providers/microsoft/azureactivedirectory/AzureActiveDirectoryCloud.java#L99-L101) ?

(This may be non-obvious, but the provided authority is [present in the AadClouds Map whether it has been validated or not](https://github.com/AzureAD/azure-activedirectory-library-for-android/blob/release/1.15.0/adal/src/main/java/com/microsoft/aad/adal/AcquireTokenRequest.java#L243) -- if validation is disabled, it will be marked as unvalidated).

As reference, this issue was discovered from the following stack trace

```java
        java.lang.NullPointerException: Attempt to invoke virtual method 'boolean com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectoryCloud.isValidated()' on a null object reference
                com.microsoft.identity.common.internal.providers.microsoft.azureactivedirectory.AzureActiveDirectoryOAuth2Strategy.getIssuerCacheIdentifier(AzureActiveDirectoryOAuth2Strategy.java:78)
                com.microsoft.identity.common.internal.cache.ADALOAuth2TokenCache.saveTokens(ADALOAuth2TokenCache.java:126)
                com.microsoft.aad.adal.TokenCacheAccessor.updateTokenCacheUsingCommonCache(TokenCacheAccessor.java:346)
                com.microsoft.aad.adal.TokenCacheAccessor.updateTokenCache(TokenCacheAccessor.java:313)
                com.microsoft.aad.adal.AcquireTokenInteractiveRequest.acquireTokenWithAuthCode(AcquireTokenInteractiveRequest.java:126)
                com.microsoft.aad.adal.AcquireTokenRequest$3.run(AcquireTokenRequest.java:849)
                java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
                java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
                java.lang.Thread.run(Thread.java:818)
```

**Comment by iambmelt** [comment]
Per @kreedula tests are now in another branch

**Pull request closed without merging**